viz/pkg/healthcheck/healthcheck.go

```diff
@@ -163,27 +164,30 @@ func (hc *HealthChecker) VizCategory() healthcheck.Category {
 			WithHintAnchor("l5d-viz-prometheus").
 			Warning().
 			WithCheck(func(ctx context.Context) error {
-				// TODO: Skip if prometheus is disabled
+				if _, ok := hc.vizNamespace.Annotations[labels.VizExternalPrometheus]; ok {
```

Comment:
For my own understanding, is this a temporary fix. I'd imagine that we would at least want to confirm that the external URL provided is reachable i.e. returning a valid Prometheus response.

Reply:
We do validate if the URL is working or not but that test is part of `viz extension self-check` where the `metrics-api` checks if the Prometheus actually works and fails if not whose output got changed a bit slightly in https://github.com/linkerd/linkerd2/pull/5665